scene/main/node.cpp

```diff
@@ -1715,8 +1715,50 @@ void Node::reparent(Node *p_parent, bool p_keep_global_transform) {
 		return;
 	}
 
+	bool preserve_owner = false;
+	Node *search = p_parent;
+	Node *owner_temp = data.owner;
+	List<Node *> common_parents;
+
+	//First determine if the original owner is an ancestor of the node we're reparenting to
+	while (search) {
+		if (search == data.owner) {
+			preserve_owner = true;
+			break;
+		}
+		search = search->data.parent;
+	}
+
+	//If so, find all children of the reparented node who have the same owner so that we can reassign them
+	if (preserve_owner) {
+		List<Node *> to_visit;
+
+		to_visit.push_back(this);
+		common_parents.push_back(this);
+
+		while (to_visit.size() > 0) {
+			Node *check = to_visit[to_visit.size() - 1];
+
+			for (int i = 0; i < check->get_child_count(); i++) {
+				Node *child = check->get_child(i, false);
+				if (child->data.owner == owner_temp) {
+					to_visit.push_back(child);
```

Comment:
You should push `to_visit` even if owner doesn't match, to handle local ancestors of editable children.